File: backend/app/schemas/agent.py
from typing import Optional, Literal, List
from datetime import datetime
from pydantic import BaseModel, Field


class AgentItem(BaseModel):
    id: int
    name: str
    code: str
    description: str
    workflow: str
    system_prompt: str
    parameters: Optional[str]
    llm_config: Optional[str]
    is_active: bool


class AgentListRequest(BaseModel):
    limit: int = 50


class AgentListResponse(BaseModel):
    total: int
    agents: List[AgentItem]


class AgentRunRequest(BaseModel):
    user_input: str
    conversation_id: int
    user_id: int
    workflow: Literal["poetry_game", "rag_chat"] = "rag_chat"
    history_limit: int = 10

File: frontend/src/types/agent.ts
export interface AgentItem {
  id: number;
  name: string;
  code: string;
  description: string;
  workflow_key: string;
  system_prompt: string;
  parameters?: string | null;
  llm_config?: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface AgentListRequest {
  limit?: number;
}

export interface AgentListResponse {
  total: number;
  agents: AgentItem[];
}

export interface AgentRunRequest {
  user_input: string;
  conversation_id: number;
  workflow?: 'poetry_game' | 'rag_chat';
  history_limit?: number;
}

export interface AgentRunResponse {
  message: string;
}
